src/solana.ts

```diff
@@ -0,0 +1,28 @@
+import type { ArcanaProvider, RequestArguments } from './provider'
+import * as bs58 from 'bs58'
+
+export class ArcanaSolanaAPI {
+  constructor(private p: ArcanaProvider) {}
+
+  async request(args: RequestArguments): Promise<unknown> {
+    const response = await this.p.request(args)
+    switch (args.method) {
+      case 'signMessage': {
+        return bs58.decode(<string>response)
+      }
```

Comment:
@mmjee response of this will be of the format

```ts
type SignatureRes = {
    publicKey: BN // can use new solanaWeb3.PublicKey(address),
    signature: Base58Decoded 
}
```